libraries/chain/block_state.cpp

```diff
@@ -60,45 +60,56 @@ block_state_ptr block_state::create_if_genesis_block(const block_state_legacy& b
    auto result_ptr = std::make_shared<block_state>();
    auto &result = *result_ptr;
 
+   // set block_header_state data ----
    result.block_id = bsp.id();
    result.header = bsp.header;
-   result.header_exts = bsp.header_exts;
-   result.block = bsp.block;
    result.activated_protocol_features = bsp.activated_protocol_features;
    result.core = finality_core::create_core_for_genesis_block(bsp.block_num());
 
+   assert(bsp.block->contains_header_extension(instant_finality_extension::extension_id())); // required by transition mechanism
+   instant_finality_extension if_ext = bsp.block->extract_header_extension<instant_finality_extension>();
+   assert(if_ext.new_finalizer_policy); // required by transition mechanism
+   result.active_finalizer_policy = std::make_shared<finalizer_policy>(*if_ext.new_finalizer_policy);
+   result.active_proposer_policy = std::make_shared<proposer_policy>();
+   result.active_proposer_policy->active_time = bsp.timestamp();
+   result.active_proposer_policy->proposer_schedule = bsp.active_schedule;
+   result.proposer_policies = {};  // none pending at IF genesis block
+   result.finalizer_policies = {}; // none pending at IF genesis block
+   result.header_exts = bsp.header_exts;
+
+   // set block_state data ----
+   result.block = bsp.block;
+   result.strong_digest = result.compute_finality_digest(); // all block_header_state data populated in result at this point
+   result.weak_digest = create_weak_digest(result.strong_digest);
+
+   // TODO: https://github.com/AntelopeIO/leap/issues/2057
+   // TODO: Do not aggregate votes on blocks created from block_state_legacy. This can be removed when #2057 complete.
+   result.pending_qc = pending_quorum_certificate{result.active_finalizer_policy->finalizers.size(), result.active_finalizer_policy->threshold, result.active_finalizer_policy->max_weak_sum_before_weak_final()};
+   result.valid_qc = {}; // best qc received from the network inside block extension, empty until first savanna proper IF block
```

Comment:
Is this just for documentation?

Reply:
Yes, Areg wanted all attributes accounted for explicitly.